openlibrary/solr/update_work.py

```diff
@@ -846,10 +852,14 @@ def solr_update(requests, debug=False, commitWithin=60000):
     :param bool debug:
     :param int commitWithin: Solr commitWithin, in ms
     """
-    h1 = HTTPConnection(get_solr())
-    url = 'http://%s/solr/update' % get_solr()
-
+    url = get_solr_base_url() + '/update'
+    parsed_url = urlparse(url)
+    if parsed_url.port:
+        h1 = HTTPConnection(parsed_url.hostname, parsed_url.port)
+    else:
+        h1 = HTTPConnection(parsed_url.hostname)
     logger.info("POSTing update to %s", url)
+    # FIXME; commit strategy / timing should be managed in config, not code
     url = url + "?commitWithin=%d" % commitWithin
```

Comment:
```suggestion
    url += "?commitWithin=%d" % commitWithin
```

Reply:
Opting out since unrelated to my changes; would like to keep this PR as small as humanly possible to just things that cause errors on solr 8 do avoid losing focus.